Luis copy button weekly summary

luisarevalo21 opened the pull request:
# Description
<img width="991" alt="Screenshot 2023-08-22 at 9 16 14 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/57dc98a9-3383-462c-9134-383f306cea2b">


## Main changes explained:

Added a copy icon to user's profile component. When viewing weekly summaries admins can quickly copy and paste their team members summaries to speed up their managerial tasks. 
Jae also suggested to add the icon to the weekly reports summary to view everyone's weekly summaries to counter-act the current bug. This way admins can copy weekly summaries there as well. 

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user or owner.
4. go to dashboard→ view profile → show weekly summary → select a team member.
5. View their weekly summaries and click on the copy button to copy their summary to your clipboard. 
<img width="899" alt="Screenshot 2023-08-22 at 9 23 55 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/ef105e5c-5f67-4c24-a2b5-450d9da500ab">


7. Verify clipboard has the exact text copied to your clipboard. 
<img width="821" alt="Screenshot 2023-08-22 at 9 11 15 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/66aeb2cc-ea7b-4a4c-82e4-7c5dd0015755">

8. Repeat from steps 3  but navigate to  dashboard→ reports → weekly summary reports  → select a team member.
9. View their weekly summaries and click on the copy button to copy their summary to your clipboard. 
<img width="692" alt="Screenshot 2023-08-22 at 9 15 21 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/457094b7-be8d-4fc2-897a-77337744632a">



## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/b13ec2c9-40a5-4fa5-b24d-c0611bb02ec3

yurisokolovicz commented on src/components/UserProfile/TeamWeeklySummaries/TeamWeeklySummaries.css:
```diff
@@ -27,3 +31,24 @@
+.team-weekly-summary-container {
+  display: flex;
+  flex-direction: row;
+  align-items: start;
```
Change for:

align-items: flex-start;

luisarevalo21 replied:
Hello @yurisokolovicz
I've fixed the code with your suggestion. 
Thanks!

yurisokolovicz replied:
Thank you very much!

yurisokolovicz reviewed (changes requested):
Dear @luisarevalo21 

The error that appeared in my console is a warning related to the CSS stylesheet file "TeamWeeklySummaries.css". The recommendation is to consider replacing the value "start" with "flex-start." This is advised because "flex-start" is the default value for aligning content within a flex container.

**Before adding flex-start:**


![Screenshot from 2023-08-23 11-23-34](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/121908988/00ed0635-aff9-49c2-aa10-f2cd73a0bfe7)


**After adding flex-start:**

![Screenshot from 2023-08-23 11-29-11](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/121908988/4333ceef-8950-45c7-a662-e319c08dbb5c)

harshidad reviewed (commented):
1. dashboard→ view profile → show weekly summary → select a team member.
      View their weekly summaries and click on the copy button to copy their summary to your clipboard.

2. navigate to dashboard→ reports → weekly summary reports → select a team member.
      View their weekly summaries and click on the copy button to copy their summary to your clipboard.

In both the scenario mentioned above, copy button working fine. Only I observed that, at the end "&nbsp" coming, even though the space, I gave in each sentence. But still in last sente
![Screenshot -1_button weekly summary #1166](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109878936/3d5c5822-7431-4736-b6eb-14c2b68bc805)
<img width="751" alt="Screenshot -2 _button weekly summary #1166" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109878936/692cb8e2-0bb8-45cf-b5b5-ba0720f80eb0">
<img width="955" alt="Screenshot -3_button weekly summary #1166" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109878936/0185ab9c-dcc0-4d0f-bb19-44e945defb9a">
nce additional text display &nbsp. 

Please check the attched screenshot.

iTvX reviewed (changes requested):
I checked you pr, copy paste looks fine.
![1](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/7c90b7da-888e-410e-8b3d-12818eab1104)

The title bar does not display all the buttons. I have tried the development branch, and this issue is not present.
![CleanShot 2023-08-23 at 13 44 05@2x](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/22ee8b34-cc31-4915-ae7f-46bf0e19af41)

DavidC0126 reviewed (commented):
Hi @luisarevalo21 I tested your PR and it works as intended, just the problem of other links not visible as an ADMIN user (not related to this PR).
<img width="1676" alt="copy_report" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/74082492/d203bb1e-f9a3-4835-9a93-5544dc78e0e6">

luisarevalo21 commented:
> I checked your pr, copy paste looks fine. ![1](https://user-images.githubusercontent.com/46791134/262794094-7c90b7da-888e-410e-8b3d-12818eab1104.png)
> 
> The title bar does not display all the buttons. I have tried the development branch, and this issue is not present. ![CleanShot 2023-08-23 at 13 44 05@2x](https://user-images.githubusercontent.com/46791134/262794233-22ee8b34-cc31-4915-ae7f-46bf0e19af41.png)

Hello @iTvX 

What type account are you using? 
As when I login as an admin or an owner, I can view everything.
If you are logging into as a volunteer, volunteers do not have access to the "suggest changes on a task" 
I logged in via the development branch and saw this. 

See screenshot below. 
<img width="889" alt="Screenshot 2023-08-23 at 7 48 38 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/bec21e9f-ebfe-4558-a10c-d6bb05d05e6a">
 

<img width="1609" alt="Screenshot 2023-08-23 at 7 44 58 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/cc415590-1029-45ff-b12d-5be8a12277fa">

Haoj1 reviewed (changes requested):
Copy the summaries in profile seems to be good. However, there is a memory leak error when I trying to open weekly summary reports page and a non reference error by FontAwesome. See the screenshots.

<img width="752" alt="1182 4" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/2c38b0c4-df00-4086-8f4c-24b82cdcca95">
<img width="752" alt="Screen Shot 2023-08-24 at 12 03 33 AM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/212cfe3f-d3f1-49e0-be74-cf15617db729">


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/51338bcf-7165-4027-a90a-fa2f41893f54

veronicacheng2 reviewed (commented):
Hi @luisarevalo21, I have tried testing the PR. Unfortunately, I can only test the copy function from the user profile because currently the other links have disappeared.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95119686/65971e6a-fd1e-4c29-ae95-7a968dcfa8a1

belagaam reviewed (dismissed):
Tested the PR as per the instructions. Everything works as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47104198/e2217adb-14b3-4179-8260-ac9b035a9973

Jiyuanxia commented:
Hi @luisarevalo21 , I have reviewed your pull request and tested it. Copy button works as you intended but I met some problems when I open weekly summaries report page.
![Screenshot 2023-08-24 171357](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92963742/e9f50264-fc56-4e30-b792-30dbfea4d4d3)
![Screenshot 2023-08-24 172019](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92963742/36b610c8-9081-42dc-af0a-8dad7b08808d)

tsunami776 commented:
Tested with Admin and Owner account. The copy functionality works well on the user's profile. I can see the 'Weekly Summaries Report' button, however, the page crashes when I click on the button, and the browser console gives a memory leak error. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/2bb5d3a5-9bc3-485d-a0b2-741fcfe0f1fe

luisarevalo21 commented:
> Copy the summaries in profile seems to be good. However, there is a memory leak error when I trying to open weekly summary reports page and a non reference error by FontAwesome. See the screenshots.
> 
> <img alt="1182 4" width="752" src="https://user-images.githubusercontent.com/99279909/262868553-2c38b0c4-df00-4086-8f4c-24b82cdcca95.png"> <img alt="Screen Shot 2023-08-24 at 12 03 33 AM" width="752" src="https://user-images.githubusercontent.com/99279909/262868555-212cfe3f-d3f1-49e0-be74-cf15617db729.png">
>  1182.mov

Fixed! Please try testing again. Thanks

luisarevalo21 commented:
> Tested with Admin and Owner account. The copy functionality works well on the user's profile. I can see the 'Weekly Summaries Report' button, however, the page crashes when I click on the button, and the browser console gives a memory leak error.
> 
>  PR.Review.1166.mp4

Fixed! Please try testing again. Thanks

luisarevalo21 commented:
> Hi @luisarevalo21 , I have reviewed your pull request and tested it. Copy button works as you intended but I met some problems when I open weekly summaries report page. ![Screenshot 2023-08-24 171357](https://user-images.githubusercontent.com/92963742/263109719-e9f50264-fc56-4e30-b792-30dbfea4d4d3.png) ![Screenshot 2023-08-24 172019](https://user-images.githubusercontent.com/92963742/263109721-36b610c8-9081-42dc-af0a-8dad7b08808d.png)

Hello, I adjusted the code and fixed it. Please retest it again. Thanks

luisarevalo21 commented:
> Hi @luisarevalo21, I have tried testing the PR. Unfortunately, I can only test the copy function from the user profile because currently the other links have disappeared.
> 
>  HGN.APP.-.Google.Chrome.2023-08-23.23-39-56.mp4

Hello @veronicacheng2  
This was an ongoing bug that occured on Monday. Please switch to another branch that was posted before Monday and try switching to this branch after. 
If that doesn't work. I had to restore my entire development env and now it works. Feel free to reach out. 
Thanks.

luisarevalo21 commented:
> Copy the summaries in profile seems to be good. However, there is a memory leak error when I trying to open weekly summary reports page and a non reference error by FontAwesome. See the screenshots.
> 
> <img alt="1182 4" width="752" src="https://user-images.githubusercontent.com/99279909/262868553-2c38b0c4-df00-4086-8f4c-24b82cdcca95.png"> <img alt="Screen Shot 2023-08-24 at 12 03 33 AM" width="752" src="https://user-images.githubusercontent.com/99279909/262868555-212cfe3f-d3f1-49e0-be74-cf15617db729.png">
>  1182.mov

Hello ! I fixed the issue that occurred in FormattedReportsJSX
As for the memory leak it occurs sometimes but is unrelated to this PR.

Please retest the code to make Formmated Reports does not crash.
Thanks

n0vay commented:
Unable to test dues to missing dashboard icons error.

liaoyufu-leo reviewed (approved):
Hi the copy button shows correct on my device! @luisarevalo21 
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23627067/826d7e46-e05a-40b5-ba34-029018ee7f75)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23627067/899c595c-44aa-4d56-a607-8b3bc9da7423)

luisarevalo21 commented:
> Unable to test dues to missing dashboard icons error.

Hello can you elaborate a bit more on the error?
Are you referencing to the the missing links on the navbar?

Thanks

arcan9 commented:
> If that doesn't work. I had to restore my entire development env and now it works. Feel free to reach out.
> Thanks.

HI @luisarevalo21. I'm also trying to test this but unable to due to the missing navbar links. By restoring the entire development environment, do you mean just re-cloning the HighestGoodNetwork repo?

yaow62023 reviewed (dismissed):
I tested your pull request, and the copy text function works properly under both the "View Profile" and "Weekly Summary Report" options. The copy buttons are also functioning correctly. However, I also have a question: I noticed that there are "&nbsp;" occurrences for spaces and line breaks. Is there a way to prevent this from happening? 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/132e4a08-51d4-497c-8f3b-25604b96256a

yurisokolovicz reviewed (approved):
Thank you!

tsunami776 reviewed (dismissed):
The Weekly Summaries Report page loads correctly without crashing. The copy function works on both the user's profile and the Weekly Summaries Report.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/9a5802b1-4a79-4a1b-aa89-8bea087d8117

Indujaks reviewed (approved):
Tested the changes, copy function works perfectly, Good Job!

<img width="1427" alt="Screenshot 2023-08-30 at 4 03 59 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/11251850/fc56fc72-ce10-40c4-bcd3-058d266f342b">

<img width="1404" alt="Screenshot 2023-08-30 at 4 05 16 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/11251850/734d8d85-c3bc-4a8d-98d8-bef25c71eee3">

beblicarl commented:
The weekly summary report page took so long to load however this functionality works as intended

[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2023.09.04-11_00_47.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/68056104/2da3c9ee-2719-46f1-81fd-7e623218054c)

one-community commented:
Thank you all! Resolve merge conflicts and I'll merge this one!

GaryB93 reviewed (changes requested):
Is this branch still under review? I tested the functionality of the copy button and it works as intended. I did notice on the Weekly Summaries Report page when you copy a summary it adds "&nbsp" to the end of a line. Could be that specific summary, as when I tested the button on my own summary on the user profile page I didn't get any "&nbsp".

Also I have some concerns with some of the file changes starting with some files being added back in to the eslintignore and prettierignore files. Seems like this branch has had a lot of changes and just want to make sure it has the most up-to-date development branch changes.

![Screenshot 2023-09-07 at 2 39 14 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108501006/c13f0270-3827-4987-b713-16e7053e0f4d)
![Screenshot 2023-09-07 at 3 19 40 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108501006/77559d9d-9324-448e-ad61-84b921c00871)

Shubhu0 commented:
Hi @luisarevalo21 I tested and reviewed your PR. It worked perfectly and displayed the summary as well as presented the copy button for the summary. Another thing to note would be that the page had a huge buffer period and took a while to load.

![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56580102/be64c04e-917f-4f08-ad2e-8674f9a50d40)

JeffLi117 reviewed (changes requested):
### Results using profile method

Report to be copied:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/334129fe-64ba-4f73-af1c-e78b0b2ea561)

Toast once copied:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/965c2ce7-ba82-4df4-839d-cc14f68811f3)

Text snippet pasted into Word doc:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/31229f8c-8a2e-43fb-ad32-087cc6d03453)

### Results using reports method

Report to be copied:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/fcc0d916-7042-44d5-8fc5-f872de95a39c)

Toast once copied:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/b3433b1b-6a5c-48b1-9651-f68bc11e5b4b)

Text snippet pasted into Word doc:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/9ba08303-ccc2-456c-a819-388413304d78)
(*Note, I also get the exact same output copying & pasting through the view profile pathway - I just chose a different summary initially)

### Changes requested
Using both methods you described in the PR, you can see that copying text that has a whitespace at the end of the line leads to pasting a "&nbsp;" at the end - this will probably slow down the work flow, which will be detrimental to the purpose of this PR. 

Adding console log within the (hopefully correct) component:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/f476b5a5-e825-4590-991e-9cb9b165710d)

The console:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112568110/19512b5a-b70e-4373-a446-e897a5933669)

This may be outside the scope of the files that you changed. It could be the way the data is being presented/parsed before copying. Nevertheless, it seems to be an issue worth addressing before merging.

JeffLi117 reviewed (dismissed):
@luisarevalo21 my mistake, I realized the issues I brought up were only because I used Word; no problem copying & pasting into Notepad. Works as intended, good job!
